### ISSUE TITLE
RuntimeError: No active exception to reraise

### ISSUE THREAD

**Issue by tehspaceg** [tweepy/tweepy#570]
```
Traceback (most recent call last):
  File "./python_script.py", line 80, in <module>
    stream.filter(track=['#warcraft'], languages=['en'])
  File "/usr/local/lib/python3.4/dist-packages/tweepy/streaming.py", line 428, in filter
    self._start(async)
  File "/usr/local/lib/python3.4/dist-packages/tweepy/streaming.py", line 346, in _start
    self._run()
  File "/usr/local/lib/python3.4/dist-packages/tweepy/streaming.py", line 286, in _run
    raise
RuntimeError: No active exception to reraise
```

Receiving this error occasionally with my script that filters the Twitter stream and looks for specific words. The failure line is:

```
stream.filter(track=['#keyword'], languages=['en'])
```

A user on StackOverflow identified this as due to how an exception is being handled: http://stackoverflow.com/a/28871066

**Comment by nikraveshucb** [comment]
changing "raise" to "raise exception" won't solve the problem.

I have not have any problem till tonight, that I started to getting the error.

Is there a fix for this?

**Comment by tehspaceg** [comment]
It doesn't fix the exceptions being thrown, but it will at least make it so an exception is raised properly for further troubleshooting.

**Comment by markwaddle** [comment]
I changed my local copy to raise the exception and it works as expected.

This occurs when an exception is raised in the stream listener (in on_data in my case). Without re-raising the exception you can't handle it or access it's stack trace.

**Comment by deusofnull** [comment]
Sorry to jump on this train late, but I am also experiencing this issue.  

How do I go about fixing it?  Do I have to change the tweepy source?

**Comment by conorgriffin** [comment]
I'm getting this issue also, nobody assigned to look at it yet it seems...

**Comment by deusofnull** [comment]
@conorgriffin   I found a way to solve this.  Either run with python 3.4 or change "raise" to "raise exception" in this code block
```
  if exception:
            # call a handler first so that the exception can be logged.
            self.listener.on_exception(exception)
            raise
```
https://github.com/tweepy/tweepy/blob/master/tweepy/streaming.py#L283

I found that solution from the link posted on top of this thread: http://stackoverflow.com/questions/28869442/unraised-exception-using-tweepy-and-mysql/28871066#28871066

If you're running on linux, change the tweepy streaming.py file found here: /usr/local/lib/python3.4/dist-packages/tweepy/streaming.py
Change python3.4 to python if your using default python install for ubuntu. 

It's a very simple fix, I thought it would be harder, but it was just making that one change.

Hope that helps!

**Comment by blckmstr** [comment]
Hey I changed the raise to raise Exception but still it comes up with the same error. 

This exception however only comes up when i am trying to write to file.

**Comment by rgduncan** [comment]
I have the same issue. I changed the above to 'raise exception' and I'm still getting the error but only when trying to write to a file. Anyone have an idea for a work around?

```
class TweetWriteListener(StreamListener):       
    
    def __init__(self,data_file_name=None):
        super(TweetWriteListener, self ).__init__()
        self.data_file_name = data_file_name

    def on_status(self, status):
        print(status)
        with open(self.data_file_name,'a') as output_file:
           output_file.write(status)
        return True

    def on_error(self, status):
        print(status)
```

**Comment by ElaineResende** [comment]
Same thing happening with me, just when I try to write to a file. Does anybody know what would be a solution?

**Comment by alallier** [comment]
Issue happening to me also

**Comment by Aaron1011** [comment]
Fixed in https://github.com/tweepy/tweepy/commit/37e318befb56f84fe8ed4f0999210e3c5d9266f6.

**Comment by Anubhav814** [comment]
Happened with #me_too (- _ -)

**Comment by Harmon758** [comment]
As mentioned, this should now be fixed with https://github.com/tweepy/tweepy/commit/37e318befb56f84fe8ed4f0999210e3c5d9266f6, which was released as part of Tweepy v3.6.0.
